slam/alarm_configuration_widget.py

```diff
@@ -150,6 +150,9 @@ def __init__(self, alarm_item: AlarmItem, kafka_producer: KafkaProducer,
             self.filter_layout = QHBoxLayout()
             self.filter_layout.addWidget(self.filter_label)
             self.filter_layout.addWidget(self.filter_edit)
+            if self.filter_edit.text():
```

Comment:
can check if item already has filter here because the edit gets populated on line 70: self.filter_edit = QLineEdit(alarm_item.alarm_filter or '')

think this is separate from the other "re-enabling issue", since guessing that issue seems to be with populating the alarm_item b4 passing to the config constructor